### PR TITLE
[Python] Add masonite

### PR DESCRIPTION
**Pull request by vaibhavmule**
Before submitting your PR, please review the following checklist :

## If you are adding a framework

+ [x] Please check configuration files
   + [x] `neph.yml`, for [neph](http://tbrand.github.io/neph) job processing
   + [x] `FRAMEWORKS.yml`, for **frameworks** list
   + [x] `travis.yml`, for CI
+ [x] A `Dockerfile` exists ? no
+ [x] All tests passes ? yes
~~~
export FRAMEWORK=<MY_FRAMEWORK>
shards install
shards build --release
bin/neph ${FRAMEWORK}
crystal spec
~~~

### CODE REVIEW

File: FRAMEWORKS.yml [hunk 1]
```diff
@@ -74,6 +74,10 @@ node:
     type: experimental
     language: 11.14
 python:
+  masonite:
+    language: "3.7"
+    version: "2.1.27"
```
Comment (waghanza) [inline FRAMEWORKS.yml]: Could you use `X.`Y` format ?

File: python/masonite/app/http/middleware/AuthenticationMiddleware.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+"""Authentication Middleware."""
```
Comment (waghanza) [inline python/masonite/app/http/middleware/AuthenticationMiddleware.py]: Are **middlewares** required for this setup ?

File: python/masonite/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+masonite>=2.1.0,<2.2.0
```
Comment (waghanza) [inline python/masonite/requirements.txt]: I find the constraint `masonite==2.1.*` more readable

**Comment by vaibhavmule** [comment]
Yes I can make number only,  currently it is string.

**Comment by waghanza** [comment]
Sure, I had use strings in the past, but it could be converted in digits

File: FRAMEWORKS.yml (hunk 1 above)
Comment (vaibhavmule) [reply FRAMEWORKS.yml]: I could but will  miss the few updates.

File: python/masonite/requirements.txt (hunk 3 above)
Comment (vaibhavmule) [reply python/masonite/requirements.txt]: I can add that.

File: python/masonite/config/middleware.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+from app.http.middleware.LoadUserMiddleware import LoadUserMiddleware
+from app.http.middleware.VerifyEmailMiddleware import VerifyEmailMiddleware
+
+"""HTTP Middleware
+HTTP middleware is middleware that will be ran on every request. Middleware
+is only ran when a HTTP call is successful (a 200 response). This list
+should contain a simple aggregate of middleware classes.
+"""
+
+HTTP_MIDDLEWARE = [
```
Comment (vaibhavmule) [inline python/masonite/config/middleware.py]: I can remove or comment out unused middleware

File: FRAMEWORKS.yml (hunk 1 above)
Comment (waghanza) [reply FRAMEWORKS.yml]: There will be tracked 😙

You can add a line on `.dependabot/config.yml`

Comment (vaibhavmule) [reply FRAMEWORKS.yml]: Okay, cool

File: python/masonite/app/http/middleware/AuthenticationMiddleware.py (hunk 2 above)
Comment (vaibhavmule) [reply python/masonite/app/http/middleware/AuthenticationMiddleware.py]: Not really, i can remove them.

File: python/masonite/config/middleware.py (hunk 4 above)
Comment (vaibhavmule) [reply python/masonite/config/middleware.py]: @josephmancuso do you have any suggestion making this setup leaner?
- I have commented out the middleware. 
- I have commented out third-party providers

Following are simple controller method which will be tested against the benchmarks. 

https://github.com/the-benchmarker/web-frameworks/pull/1249/files#diff-5f3a0e2d514d4ef3d3c2e62feb030f9eR5

**Comment by vaibhavmule** [comment]
@waghanza I almost forgot about it, I have made the changes you can start a review on.

File: python/masonite/config/middleware.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    # LoadUserMiddleware,
+    # CsrfMiddleware,
+    # ResponseMiddleware,
```
Comment (josephmancuso) [inline python/masonite/config/middleware.py]: If this benchmark is returning responses with the proper headers then the `ResponseMiddleware` needs to be uncommented

File: python/masonite/config/application.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+NAME = env('APP_NAME', 'Masonite 2.1')
+
+"""Application Debug Mode
+When your application is in debug mode, detailed error messages with
+stack traces will be shown on every error that occurs within your
+application. If disabled, a simple generic error page is shown
+"""
+
+DEBUG = env('APP_DEBUG', False)
```
Comment (josephmancuso) [inline python/masonite/config/application.py]: Just make sure debug is `False` when running this in production for a better speed boost

Comment (vaibhavmule) [reply python/masonite/config/application.py]: I'm not sure about how environment variable going to work,  need to ask @waghanza

File: python/masonite/config/middleware.py (hunk 5 above)
Comment (vaibhavmule) [reply python/masonite/config/middleware.py]: I didn't know how did I missed it I will add it. Thanks for the review

Comment (vaibhavmule) [reply python/masonite/config/middleware.py]: I added it back.

File: python/masonite/config/application.py (hunk 6 above)
Comment (waghanza) [reply python/masonite/config/application.py]: by default, there is not
so having a default value to `False` is :ok:

Comment (josephmancuso) [reply python/masonite/config/application.py]: awesome 👌

**Comment by waghanza** [comment]
@vaibhavmule @josephmancuso Calling `/` with `curl` I have :
~~~python
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/whitenoise/base.py", line 75, in __call__
    return self.application(environ, start_response)
  File "/home/waghanza/workspace/benchmark/web/python/masonite/bootstrap/start.py", line 40, in app
    container.make('ExceptionHandler').load_exception(e)
  File "/usr/local/lib/python3.7/site-packages/masonite/exception_handler.py", line 58, in load_exception
    self.handle(exception)
  File "/usr/local/lib/python3.7/site-packages/masonite/exception_handler.py", line 67, in handle
    request.status(500)
  File "/usr/local/lib/python3.7/site-packages/masonite/request.py", line 307, in status
    self.app().bind('StatusCode', text_status)
AttributeError: 'NoneType' object has no attribute 'bind'
~~~

**Comment by josephmancuso** [comment]
issue is fixed in 2.1.29. I believe it was how the mienheld worker was constructing the environment. 

just give it 5 min or so for the release to fully go throughh CI and propagate.

**Pull request merged**